### PR TITLE
fix: resolve import collisions for enums

### PR DESCRIPTION
**Pull request by doochik**
Similar to #339. Also fixes generated functions

Sorry, my fault. I've tested collisions for types and haven't tested for JSON methods

### CODE REVIEW

File: src/enums.ts [hunk 1]
```diff
@@ -108,7 +110,9 @@ export function generateEnumToJson(fullName: string, enumDesc: EnumDescriptorPro
 /** Generates a function with a big switch statement to encode our string enum -> int value. */
 export function generateEnumToNumber(fullName: string, enumDesc: EnumDescriptorProto): Code {
   const chunks: Code[] = [];
-  chunks.push(code`export function ${camelCase(fullName)}ToNumber(object: ${fullName}): number {`);
+
+  const functionName = camelCase(fullName) + 'ToNumber';
+  chunks.push(code`export function ${def(functionName)}(object: ${fullName}): number {`);
```
Comment (stephenh) [inline src/enums.ts:115]: Ahh, interesting wrinkle. Makes sense!

**Review by stephenh — APPROVED** [review]

**Comment by doochik** [comment]
@stephenh What I need to release new version? Just merge it to `main`?

**Comment by stephenh** [comment]
> What I need to release new version? Just merge it to main?

@doochik Yep! I was kind of a late adopter, but a coworker got me hooked on the semantic-release tool suite that does a pretty great job of automating releases, in terms of doing tags / changelog / github release / npm upload / etc.

**Comment by doochik** [comment]
> > What I need to release new version? Just merge it to main?
> 
> @doochik Yep! I was kind of a late adopter, but a coworker got me hooked on the semantic-release tool suite that does a pretty great job of automating releases, in terms of doing tags / changelog / github release / npm upload / etc.

Thanks!

**Pull request merged**